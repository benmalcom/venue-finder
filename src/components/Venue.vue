<template>
    <div class="col-md-4 col-sm-6">
        <div class="row venue-item col-sm-12 col-12 text-right shadow">
            <div class="col-md-3 col-3 center-content">
                <div class="ml-auto circle circle-md center-content business-image-placeholder">
                    <img src="images/venue-marker.png" class="full-height full-width">
                </div>
            </div>
            <div class="col-md-9 col-9 center-content pl-0">
                <div class="mr-auto text-left">
                    <h6 class="mb-1 mt-1 text-md">
                        <strong>
                            Victoria's Secret
                        </strong>
                    </h6>
                    <h6 class="text-sm text-muted">
                        174/194 Obafemi Awolowo way,
                        Alausa, Ikeja, Lagos state.
                    </h6>
                    <span class="badge badge-warning text-white">SALON</span>
                </div>
            </div>
        </div>
    </div>
</template>

<script>
	export default {
		name: 'Venue'
	}
</script>

<style scoped>

</style>
