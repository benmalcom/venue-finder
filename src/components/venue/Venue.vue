<template>
    <div class="col-lg-4 col-md-4 col-sm-6">
        <div class="row venue-item shadow-lite">
            <div class="col-md-4 col-4 full-height full-width p-0">
                    <img src="../../assets/images/venue-marker.png" class="marker-image full-height full-width">
            </div>
            <div class="col-md-8 col-8 pt-30">
                <div class="mr-auto text-left">
                    <h6 class="mb-1 mt-1">
                        <strong class="name">
                            {{venue.name.length > 40 ? `${venue.name.substring(0,37)}...` : venue.name}}
                        </strong>
                    </h6>
                    <h6 class="text-sm text-muted address">
                        {{getFormattedAddress(venue.location)}}
                    </h6>
                    <label class="badge badge-secondary text-white text-xs category" v-if="getCategory(venue)">
                        <i class="fa fa-tags"></i> {{ getCategory(venue) }}
                    </label>
                    <label class="badge badge-warning text-white text-xs category" v-else><i class="fa fa-tags"></i> Not Available</label>

                </div>
            </div>
        </div>
    </div>
</template>

<script>
    import { getCategory, getFormattedAddress } from '../../helpers';
	export default {
		name: 'Venue',
        props: ['venue'],
        data(){
			return {
			    category: null,
                address: null,
            };
        },
        methods: {
			getFormattedAddress,
            getCategory
        }
	}
</script>
