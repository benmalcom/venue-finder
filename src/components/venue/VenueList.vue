<template>
    <div class="col-md-11 mx-auto mt-30 pl-1">
        <a class="btn btn-sm btn-filter btn-outline-secondary d-sm-none mb-3"
           data-toggle="offcanvas">
            <i class="fa fa-filter"></i> Filter</a>
        <h5 class="ml-2" v-if="venues && venues.length"><strong>Available Venues</strong></h5>
        <div class="row p-0">
            <Venue v-for="(value,key) in venues" :venue="value" :key="key"></Venue>
        </div>
    </div>
</template>

<script>
    import Venue from './Venue.vue';
    import $ from 'jquery';
	export default {
		name: 'VenueList',
        props: ['venues'],
		mounted() {
			// Bind mobile toggling click events for filter
			$('[data-toggle=offcanvas]').click(() => $('.offcanvas').toggleClass('active'));
			$('.close-filter').click(() => $('.offcanvas').toggleClass('active'));
		},
        components: {Venue}
	}
</script>

<style scoped>

</style>
