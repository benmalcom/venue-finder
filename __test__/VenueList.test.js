import { shallow } from 'vue-test-utils'
import VenueList from '../src/components/venue/VenueList.vue';
import Venue from '../src/components/venue/Venue.vue';
import venueObj from './venue.json';

describe('Component: VenueList.vue', () => {
	let comp;

	beforeEach(() => {
		comp = shallow(VenueList, { // Create a shallow instance of the component
			propsData: {
				venues: [venueObj]
			}
		});
	});

	it('Has the expected html structure', () => {
		expect(comp.element).toMatchSnapshot();
	});

	it('Has :venues prop to be set', () => {
		expect(comp.props().venues).toBeDefined();
	});

	it('Expects :venues to be an array with length 1', () => {
		const venues = comp.props().venues;
		expect(Array.isArray(venues)).toBeTruthy();
		expect(venues.length).toEqual(1);
	});

	it('Should contain Venue component', () => {
		expect(comp.contains(Venue)).toBeTruthy();
	});

});
