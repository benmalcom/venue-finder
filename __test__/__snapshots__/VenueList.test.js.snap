// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: VenueList.vue Has the expected html structure 1`] = `
<div
  class="col-md-11 col-sm-12 mx-auto mt-30 pl-1"
>
  <a
    class="btn btn-sm btn-filter btn-outline-secondary d-sm-none mb-3"
    data-toggle="offcanvas"
  >
    <i
      class="fa fa-filter"
    />
     Filter
  </a>
   
  <h5
    class="ml-2"
  >
    <strong>
      Available Venues
    </strong>
  </h5>
   
  <div
    class="row p-0"
  >
    <!---->
  </div>
</div>
`;
